recipes/opentdf-client/all/conanfile.py

```diff
@@ -36,63 +39,98 @@ def _minimum_cpp_standard(self):
-                self.name, self.settings.compiler))
+            self.output.warn(f"{self.name} recipe lacks information about the {self.settings.compiler} compiler support.")
         else:
-            if tools.Version(self.settings.compiler.version) < min_version:
-                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
-                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+            if Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(f"{self.name} requires C++{self._minimum_cpp_standard} support."
+                                                 "The current compiler {self.settings.compiler} {self.settings.compiler.version} does not support it.")
 
     def requirements(self):
-        self.requires("openssl/1.1.1l@")
-        self.requires("boost/1.76.0@")
-        self.requires("zlib/1.2.11@")
-        self.requires("ms-gsl/2.1.0@")
-        self.requires("libxml2/2.9.10@")
-        self.requires("libarchive/3.5.1@")
-        self.requires("nlohmann_json/3.10.4@")
-        self.requires("jwt-cpp/0.4.0@")
+        self.requires("openssl/1.1.1q")
+        self.requires("boost/1.76.0")
+        self.requires("ms-gsl/2.1.0")
+        self.requires("libxml2/2.9.14")
+        self.requires("nlohmann_json/3.11.1")
+        self.requires("jwt-cpp/0.4.0")
+        if not self.options.without_libarchive:
+            self.requires("libarchive/3.6.1")
+        # We do not require libiconv but conan-center only allows 'stock' references, and boost+libxml2
+        # specify differerent versions, which causes a build fail due to the dependency conflict.
+        # Overriding the version here allows a clean build with the stock build settings.
+        if not self.options.without_libiconv:
+            self.requires("libiconv/1.17@")
+        # Same for zlib
+        if not self.options.without_zlib:
+            self.requires("zlib/1.2.12@")
 
     def config_options(self):
+        # Do not require libarchive as of 1.1.0
+        if Version(self.version) < "1.1.0":
+            self.options.without_libarchive = False
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.options.without_zlib:
+            self.options["libxml2"].zlib = False
+        if self.options.without_libiconv:
+            self.options["boost"].without_locale = True
+            self.options["boost"].without_log = True
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
```

Comment:
```suggestion
        for data in self.conan_data.get("patches", {}).get(self.version, []):
            patch(self, **patch)
```

Reply:
Updated

Reply:
Also needed to change to `patch(self, **data)`